src/charm.py

```diff
@@ -1186,12 +1239,22 @@ def _plugin_reconciliation(self):
         and adjust plugin options for these three plugins according to charm config.
         """
         self._addon_reconciliation("plugin")
+        self._plugin_swift_reconciliation()
```

Comment:
Could you explain why you moved this call outside the `is_leader` ?

Reply:
Some steps of the `_plugin_swift_reconciliation` requires to run every unit, not just the leader unit, so move the `_plugin_swift_reconciliation` outside `is_leader`. And in `_plugin_swift_reconciliation`, update the code to run different processes in leader and non-leader units.